File: reducing-technical-debt-michael-feathers-course/techdebt/cpp/scheduler/SchedulerDisplay.cpp


#include "SchedulerDisplay.h"
#include <climits>
#include <cstdio>
#include <cstdlib>
#include <cctype>


void SchedulerDisplay::ShowEvent(Event *event)
{
	for (int n = 0; n < INT_MAX / 128; ++n) {
		int c = rand();
		if (c > 0 && c <= UCHAR_MAX && (isalpha(c) || ispunct(c) || isdigit(c)))
			printf("%c", c);
	}
}

File: reducing-technical-debt-michael-feathers-course/techdebt/cs/scheduler/SchedulerDisplay.cs
using System;

namespace SchedulerCSharp
{
	public class SchedulerDisplay 
	{
		public virtual void ShowEvent(Event anEvent) 
		{
			for(int n = 0; n < 1000; n++) 
			{
				Console.WriteLine("[" + anEvent.Date + "]");	
			}
		}
	}
}
